gobblin-core/src/main/java/org/apache/gobblin/converter/json/JsonStringToJsonIntermediateConverter.java

```diff
@@ -76,40 +80,495 @@ public JsonArray convertSchema(String inputSchema, WorkUnitState workUnit) throw
     if (!this.unpackComplexSchemas) {
       return new SingleRecordIterable<>(inputRecord);
     }
+    JsonObject rec = parseJsonBasedOnSchema(inputRecord, outputSchema);
+    return new SingleRecordIterable(rec);
+  }
 
-    JsonObject outputRecord = new JsonObject();
+  /**
+   * Overloads parseJsonBasedOnSchema to support JsonArray based inputs
+   * @param input
+   * @param schema
+   * @return
+   * @throws DataConversionException
+   */
+  private JsonArray parseJsonBasedOnSchema(JsonArray input, JsonArray schema)
+      throws DataConversionException {
+    JsonObject firstSchema = schema.get(0).getAsJsonObject().get("dataType").getAsJsonObject();
```

Comment:
I think get("dataType").getAsJsonObject() can be replaced by one of your helper functions, and I see several other places where the helper could be used as well.

Reply:
Instead of having helper function within JsonUtils. I have created a class JsonSchema which represents the schema and has all the necessary methods. This abstraction will remove redundant getAsJsonObject(), getAsJsonArray() etc which are lingering in the entire converter.